src/parser.rs

```diff
@@ -75,19 +80,30 @@ pub fn parse_command(lexer: &mut Lexer, script: &mut Script) -> Result<bool, (Pa
 			}
 			"position" => {
 				let instance = InstanceName(inline(lexer.string())?);
-				script.commands.push(Command::Position(instance, position(lexer)?, None));
+				let position = position(lexer)?;
+				let animation = animation(lexer, true)?;
+				script.commands.push(Command::Position(instance, position, animation));
 			}
 			"spawn" => {
 				let character = CharacterName(inline(lexer.string())?);
 				let state = StateName(inline(lexer.string())?);
 				let position = position(lexer)?;
-
-				script.commands.push(Command::Spawn(character, state, position,
-					match inline(lexer.token())? {
-						None | Some(Token::Terminator) => None,
-						Some(Token::String(string)) => Some(InstanceName(string)),
-						Some(_) => return Err((ParserError::UnexpectedToken, Token::Terminator)),
-					}, None));
+				let mut check_with = true;
+				let mut is_end = false;
+				let instance_name = match inline(lexer.token())? {
+					None | Some(Token::Terminator) => {
+						is_end = true;
+						None
+					}
+					Some(Token::Identifier(ident)) if ident == "with" => {
```

Comment:
Use the full variable name `identifier` instead of `ident`

Reply:
Done